std/range/package.d

```diff
@@ -8647,3 +8647,237 @@ if (is(typeof(fun) == void) || isSomeFunction!fun)
 
     auto r = [1, 2, 3, 4].tee!func1.tee!func2;
 }
+
+template CommonElementType(Rs...)
+{
+    alias CommonElementType = CommonType!(staticMap!(ElementType, Rs));
+}
+
+alias isSortedRange(R) = isInstanceOf!(SortedRange, R); // TODO Or use: __traits(isSame, TemplateOf!R, SortedRange)
+
+/**
+   Merge several sorted ranges $(D rs) with less-than predicate function $(D
+   pred) into one single sorted range containing the sorted union of the
+   elements of inputs.
+
```

Comment:
Few content ideas:

"This algorithm is lazy, doing work progressively as elements are pulled off the result."

"All of its inputs must be instantiations of SortedRange (insert xref appropriately). Use the result of `sort`, or `assumeSorted` to merge ranges known to be sorted".

"Average complexity is $(BIGOH n * k)` for $(D k) ranges of maximum length $(D n)."